component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -175,6 +191,77 @@ public boolean isValidAssertion(SignedJWT signedJWT) throws OAuthClientAuthnExce
+    public boolean isValidAssertion(SignedJWT signedJWT, boolean isBackchannelCall) throws OAuthClientAuthnException {
+
+        String errorMessage;
+
+        if (signedJWT == null) {
+            errorMessage = "No valid JWT assertion found for " + Constants.OAUTH_JWT_BEARER_GRANT_TYPE;
+            return logAndThrowException(errorMessage);
+        }
+        try {
+            JWTClaimsSet claimsSet = getClaimSet(signedJWT);
+
+            if (claimsSet == null) {
+                errorMessage = "Claim set is missing in the JWT assertion";
+                throw new OAuthClientAuthnException(errorMessage, OAuth2ErrorCodes.INVALID_REQUEST);
+            }
+
+            String jwtIssuer = claimsSet.getIssuer();
+            String jwtSubject = resolveSubject(claimsSet);
+            List<String> audience = claimsSet.getAudience();
+            Date expirationTime = claimsSet.getExpirationTime();
+            String jti = claimsSet.getJWTID();
+            Date nbf = claimsSet.getNotBeforeTime();
+            Date issuedAtTime = claimsSet.getIssueTime();
+            long currentTimeInMillis = System.currentTimeMillis();
+            long timeStampSkewMillis = OAuthServerConfiguration.getInstance().getTimeStampSkewInSeconds() * 1000;
+            OAuthAppDO oAuthAppDO = getOAuthAppDO(jwtSubject);
+            String consumerKey = oAuthAppDO.getOauthConsumerKey();
+            String tenantDomain = oAuthAppDO.getUser().getTenantDomain();
+            if (!validateMandatoryFeilds(mandatoryClaims, claimsSet)) {
+                return false;
+            }
+
+            //Validate issuer and subject.
+            if (!validateIssuer(jwtIssuer, consumerKey) || !validateSubject(jwtSubject, consumerKey)) {
+                return false;
+            }
+
+            // Get audience.
+            List<String> validAud = getValidAudience(tenantDomain, isBackchannelCall);
+            long expTime = 0;
+            long issuedTime = 0;
+            if (expirationTime != null) {
+                expTime = expirationTime.getTime();
+            }
+            if (issuedAtTime != null) {
+                issuedTime = issuedAtTime.getTime();
+            }
+
+            //Validate signature validation, audience, nbf,exp time, jti.
+            if (!validateJTI(signedJWT, jti, currentTimeInMillis, timeStampSkewMillis, expTime, issuedTime) ||
+                    !validateAudience(validAud, audience) || !validateJWTWithExpTime(expirationTime, currentTimeInMillis
+                    , timeStampSkewMillis) || !validateNotBeforeClaim(currentTimeInMillis, timeStampSkewMillis, nbf) ||
+                    !validateAgeOfTheToken(issuedAtTime, currentTimeInMillis, timeStampSkewMillis) || !isValidSignature
+                    (consumerKey, signedJWT, tenantDomain, jwtSubject)) {
+                return false;
+            }
+
+            return true;
+
+        } catch (IdentityOAuth2Exception e) {
+            return logAndThrowException(e.getMessage());
```

Comment:
Pass the exception

Reply:
A new exception is thrown using logAndThrowException method across the whole class